File: Deck.php
<?php

class Deck
{
    /**
     * @var Card[]
     */
    private $_Cards;

    public function __construct()
    {
        $this->_Cards = array();
    }

    /**
     * @param Card $Card
     * @throws DeckIntegrityException
     */
    public function add(Card $Card)
    {
        if (array_key_exists("{$Card}", $this->_Cards)) {
            throw new DeckIntegrityException('Cannot contain the same card twice!');
        } else {
            $this->_Cards["{$Card}"] = $Card;
        }
    }

    /**
     * @return int
     */
    public function size()
    {
        return count($this->_Cards);
    }

    /**
     * @return void
     */
    public function populate()
    {
        $Suits = array(
            Suit::Spades(),
            Suit::Hearts(),
            Suit::Clubs(),
            Suit::Diamonds(),
        );

        foreach ($Suits as $Suit) {
            $this->_Cards[] = Cards::aceOf($Suit);
            $this->_Cards[] = Cards::twoOf($Suit);
            $this->_Cards[] = Cards::threeOf($Suit);
            $this->_Cards[] = Cards::fourOf($Suit);
            $this->_Cards[] = Cards::fiveOf($Suit);
            $this->_Cards[] = Cards::sixOf($Suit);
            $this->_Cards[] = Cards::sevenOf($Suit);
            $this->_Cards[] = Cards::eightOf($Suit);
            $this->_Cards[] = Cards::nineOf($Suit);
            $this->_Cards[] = Cards::tenOf($Suit);
            $this->_Cards[] = Cards::jackOf($Suit);
            $this->_Cards[] = Cards::queenOf($Suit);
            $this->_Cards[] = Cards::kingOf($Suit);
        }
    }

    /**
     * @return Card[]
     */
    public function getCards()
    {
        return array_values($this->_Cards);
    }
}

File: Deck.cs
using System.Linq;
using System.Collections.Generic;

class Deck {

    private List<Card> cards;

    public Deck() {
        cards = new List<Card>();
    }

    public void add(Card card) {
        if (cards.Contains(card)) {
            throw new DeckIntegrityException("Cannot contain the same card twice!");
        } else {
            cards.Add(card);
        }
    }

    public int size() {
        return cards.Count;
    }

    public void populate() {
        var Suits = new Suit[] {
            Suit.Spades(),
            Suit.Hearts(),
            Suit.Clubs(),
            Suit.Diamonds(),
        };

        foreach (var suit in Suits) {
            cards.Add(Cards.aceOf(suit));
            cards.Add(Cards.twoOf(suit));
            cards.Add(Cards.threeOf(suit));
            cards.Add(Cards.fourOf(suit));
            cards.Add(Cards.fiveOf(suit));
            cards.Add(Cards.sixOf(suit));
            cards.Add(Cards.sevenOf(suit));
            cards.Add(Cards.eightOf(suit));
            cards.Add(Cards.nineOf(suit));
            cards.Add(Cards.tenOf(suit));
            cards.Add(Cards.jackOf(suit));
            cards.Add(Cards.queenOf(suit));
            cards.Add(Cards.kingOf(suit));
        }
    }

    public List<Card> getCards() {
        return cards;
    }
}